source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -273,7 +273,30 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     }
 
     for (const std::string& header : tracing_config.request_headers_for_tags()) {
-      request_headers_for_tags.push_back(Http::LowerCaseString(header));
+      custom_tags.emplace_back(Tracing::CustomTagType::REQUEST_HEADER, header, header);
+    }
+
+    for (const envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager::
+             Tracing::CustomTag& tag : tracing_config.custom_tags()) {
+      Tracing::CustomTagType type;
+      const std::string& default_value = tag.has_default_value() ? tag.default_value().value() : "";
+      switch (tag.type()) {
+      case envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager::
+          Tracing::CustomTag::LITERAL:
+        type = Tracing::CustomTagType::LITERAL;
+        break;
+      case envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager::
+          Tracing::CustomTag::ENVIRONMENT:
+        type = Tracing::CustomTagType::ENVIRONMENT;
+        break;
+      case envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager::
+          Tracing::CustomTag::REQUEST_HEADER:
+        type = Tracing::CustomTagType::REQUEST_HEADER;
+        break;
+      default:
+        NOT_REACHED_GCOVR_EXCL_LINE;
```

Comment:
here is translating the proto Type enum, i think `NOT_REACHED` default brach is just for what it is, also keeps same with other enum translating.
i'm not sure what would lead it to be not intended wrong.

Reply:
its not wrong, but sometimes the compiler is able to determine that the code is unreachable on its own, so no need to tell it to exclude it from coverage. leaving it is fine though, because as you say a lot of other switch cases does this